File: src/main/java/com/wgzhao/addax/admin/dto/ApiResponse.java
package com.wgzhao.addax.admin.dto;

import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

import java.util.Collections;
import java.util.List;

@Builder
@Getter
@Setter
public class ApiResponse<T> {
    private int code;
    private String message;
    private T data;

    public ApiResponse() {
    }

    public ApiResponse(int code, String message, T data) {
        this.code = code;
        this.message = message;
        this.data = data;
    }

    // 静态方法构造统一返回结果
    public static <T> ApiResponse<T> success(T data) {
        return new ApiResponse<>(0, "success", data);
    }

    public static <T> ApiResponse<T> error(int code, String message) {
        return new ApiResponse<>(code, message, null);
    }

    @Override
    public String toString() {
        return "CommResponseDto{" +
                "code=" + code +
                ", message='" + message +
                '}';
    }
}

File: src/main/kotlin/com/wgzhao/addax/admin/dto/ApiResponse.kt
package com.wgzhao.addax.admin.dto

import com.fasterxml.jackson.annotation.JsonInclude

@JsonInclude(JsonInclude.Include.NON_NULL)
data class ApiResponse<T>(
    val code: Int,
    val message: String?,
    val data: T? = null
) {
    companion object {
        fun <T> success(data: T?, message: String? = "success", code: Int = 0): ApiResponse<T> =
            ApiResponse(code = code, message = message, data = data)

        fun <T> error(code: Int, message: String? = "error"): ApiResponse<T> =
            ApiResponse(code = code, message = message, data = null)
    }
}
